cmd/lukso/validator.go

```diff
@@ -287,15 +289,68 @@ func sendDeposit(ctx *cli.Context) error {
 	return nil
 }
 
-func initValidator(ctx *cli.Context) error {
-	if ctx.String(validatorKeysDirFlag) == "" {
-		return errKeysNotProvided
+func bumpGasAndSend(eth *ethclient.Client, tx *types.Transaction, signer *bind.TransactOpts) (*types.Transaction, error) {
+	log.Warn("Transaction failed with underpriced error - resending...")
+	gas := tx.Gas() + uint64(gasBump)
+
+	log.Debugf("Bumping gas, had %d, added %d", tx.Gas(), gas)
+	gas += tx.Gas()
+
+	fee := tx.GasFeeCap()
+	feeFloat, ok := big.NewFloat(0).SetString(fee.String())
+	if ok {
+		feeFloat = feeFloat.Mul(feeFloat, big.NewFloat(1.15))
+		feeFloat.Int(fee)
+		log.Debugf("Bumping fee %s -> %s WEI", tx.GasFeeCap().String(), fee.String())
+	} else {
+		log.Warnf("Failed to bump base fee: not ok fir bigFloat construction")
 	}
+
+	timeout := time.Now().Add(time.Second * 10)
+	nonce := tx.Nonce()
+	for {
+		if time.Now().After(timeout) {
+			return nil, errors.New("failed to send transaction - timed out with handleable noncing errors")
+		}
+
+		signed, err := signer.Signer(signer.From, types.NewTx(&types.DynamicFeeTx{
+			ChainID:   tx.ChainId(),
+			Nonce:     nonce,
+			GasTipCap: tx.GasTipCap(),
+			GasFeeCap: fee,
+			Gas:       gas,
+			To:        tx.To(),
+			Value:     tx.Value(),
+			Data:      tx.Data(),
+		}))
+
+		if err != nil {
+			return nil, err
+		}
+
+		log.Debugf("Signed bumped gas tx, gas=%d hash=%s", signed.Gas(), signed.Hash().String())
+
+		err = eth.SendTransaction(context.Background(), signed)
+		if err != nil {
+			switch strings.Contains(err.Error(), errUnderpriced) {
+			case true: // ie. data race for nonce getter
```

Comment:
Don't you have logic for guaranteeing good nonces ?

Reply:
Incrementing manually seems to be working most of the time, but @mxmar stumbled upon this issue once, so I think it's better to keep any error lookups, just in case